dfs/filebase/filebase.go

```diff
@@ -0,0 +1,241 @@
+package filebase
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/SkynetLabs/go-skynet/v2"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	s3types "github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/containerish/OpenRegistry/config"
+	"github.com/containerish/OpenRegistry/dfs"
+	"github.com/containerish/OpenRegistry/types"
+	"github.com/opencontainers/go-digest"
+)
+
+type filebase struct {
+	client *s3.Client
+	bucket string
+}
+
+func New(cfg *config.S3CompatibleDFS) dfs.DFS {
+	client := dfs.NewS3Client(cfg.Endpoint, cfg.AccessKey, cfg.SecretKey)
+	return &filebase{
+		client: client,
+		bucket: cfg.BucketName,
+	}
+}
+
+func (fb *filebase) CreateMultipartUpload(layerKey string) (string, error) {
+	input := &s3.CreateMultipartUploadInput{
+		Bucket:            &fb.bucket,
+		Key:               &layerKey,
+		ACL:               s3types.ObjectCannedACLPublicRead,
+		ChecksumAlgorithm: s3types.ChecksumAlgorithmSha256,
+		ContentEncoding:   aws.String("gzip"),
+		StorageClass:      s3types.StorageClassStandard,
+	}
+	upload, err := fb.client.CreateMultipartUpload(context.Background(), input)
+	if err != nil {
+		return "", err
+	}
+
+	return *upload.UploadId, nil
+}
+
+func (fb *filebase) UploadPart(
+	ctx context.Context,
+	uploadId string,
+	layerKey string,
+	digest string,
+	partNumber int64,
+	content io.ReadSeeker,
+	contentLength int64,
+) (s3types.CompletedPart, error) {
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*20)
```

Comment:
so as discussed, let's change it to 10 minutes since the database txn is going to expire in 10 minutes